### PR TITLE
Make superuser owner of metric tables and views

### PR DESCRIPTION
**Pull request by jgpruitt**
## Description

This branch corresponds to the extension PR https://github.com/timescale/promscale_extension/pull/152

A number of tests needed to be changed. New functions were added to the extension wrapping timescaledb functions due to privilege changes. The corresponding tests needed to be pointed to the new wrapper functions.

## Merge requirements

Please take into account the following non-code changes that you may need to make with your PR:

- [x] CHANGELOG entry for user-facing changes
- [x] Updated the relevant documentation

### CODE REVIEW

File: pkg/tests/upgrade_tests/upgrade_test.go [hunk 1]
```diff
@@ -83,7 +85,7 @@ func getDBImages(extensionState testhelpers.TestOptions) (prev string, clean str
 	if err != nil {
 		panic("unable to get docker image version")
 	}
-	return "timescaledev/promscale-extension:0.1.2-ts2-" + pgVersion, dockerImageName, nil
+	return "timescaledev/promscale-extension:0.3.2-2.6.0-pg" + pgVersion, dockerImageName, nil
```
Comment (cevian) [inline pkg/tests/upgrade_tests/upgrade_test.go:88]: why did we need this change again?

File: pkg/tests/upgrade_tests/upgrade_test.go [hunk 2]
```diff
@@ -40,8 +40,10 @@ import (
 )
 
 var (
-	testDatabase       = flag.String("database", "tmp_db_timescale_upgrade_test", "database to run integration tests on")
-	printLogs          = flag.Bool("print-logs", false, "print TimescaleDB logs")
+	testDatabase = flag.String("database", "tmp_db_timescale_upgrade_test", "database to run integration tests on")
+	printLogs    = flag.Bool("print-logs", false, "print TimescaleDB logs")
+	// use "local/dev_promscale_extension:head-ts2-pg13" for local testing
+	dockerImage        = flag.String("image", "ghcr.io/timescale/dev_promscale_extension:develop-ts2-pg14", "docker image for database")
```
Comment (cevian) [inline pkg/tests/upgrade_tests/upgrade_test.go:46]: We may need to pass this in the cicd GitHub Actions code. Let's ask @JamesGuthrie what the plan here is.

File: pkg/tests/upgrade_tests/upgrade_test.go (hunk 1 above)
Comment (jgpruitt) [reply pkg/tests/upgrade_tests/upgrade_test.go:88]: There were a couple of timescaledb bug fixes we need that show up in 2.6.0. I don't remember at the moment what they were. Seems like one had to do with reassigning ownership of a hypertable. This is the oldest image that has 2.6.0

Comment (jgpruitt) [reply pkg/tests/upgrade_tests/upgrade_test.go]: fixed

**Review by antekresic — APPROVED** [review]
LGTM.

Would like some docs on the catalog functions in the extension but that's out of scope here.

File: .github/workflows/go-e2e.yml [hunk 3]
```diff
@@ -63,32 +109,48 @@ jobs:
...
+      uses: docker/login-action@v1
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Test ${{ matrix.test-setups.name }}
+      env:
+        DOCKER_IMAGE:  ${{  needs.pick_docker_image.outputs.docker_image_14 }}
+      run: go test -race -timeout=30m ./pkg/tests/upgrade_tests/ -timescale-docker-image=$DOCKER_IMAGE
```
Comment (JamesGuthrie) [inline .github/workflows/go-e2e.yml:152]:
```suggestion
      run: go test -race -v -timeout=30m ./pkg/tests/upgrade_tests/ -timescale-docker-image=$DOCKER_IMAGE
```

File: .github/workflows/go-e2e.yml [hunk 4]
```diff
@@ -63,32 +109,48 @@ jobs:
...
+    - name: Login to GitHub Container Registry
+      uses: docker/login-action@v1
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Test ${{ matrix.test-setups.name }}
+      env:
+        DOCKER_IMAGE:  ${{  needs.pick_docker_image.outputs.docker_image_14 }}
```
Comment (JamesGuthrie) [inline .github/workflows/go-e2e.yml:151]: These tests don't run against pg14, they're skipped.

**Review by JamesGuthrie — COMMENTED** [review]
We're also missing something to run upgrade tests locally:

```diff
diff --git a/Makefile b/Makefile
index 179c54c..e3f140e 100644
--- a/Makefile
+++ b/Makefile
@@ -8,7 +8,7 @@ build: generate
 	go build -ldflags "-X 'github.com/timescale/promscale/pkg/version.Branch=${GIT_BRANCH}' -X 'github.com/timescale/promscale/pkg/version.CommitHash=${GIT_COMMIT}'" -o dist/promscale ./cmd/promscale
 
 .PHONY: test
-test: unit e2e
+test: unit e2e upgrade-test
 
 .PHONY: unit
 unit: generate
@@ -26,6 +26,12 @@ e2e: pkg/tests/testdata/traces-dataset.sz generate
 	go test -v ./pkg/tests/end_to_end_tests/ -use-timescaledb=false -timescale-docker-image=$(DOCKER_IMAGE)
 	go test -v ./pkg/tests/end_to_end_tests/ -use-multinode -timescale-docker-image=$(DOCKER_IMAGE)
 
+.PHONY: upgrade-test
+upgrade-test: CURRENT_BRANCH?=$(shell git branch --show-current | sed 's#/#-#')
+upgrade-test: DOCKER_IMAGE?=$(shell ./scripts/fallback-docker.sh $(LOCAL_DOCKER_BASE):head-ts2-pg13 $(GHCR_DOCKER_BASE):$(CURRENT_BRANCH)-ts2-pg13 $(GHCR_DOCKER_BASE):develop-ts2-pg13)
+upgrade-test:
+	go test -v ./pkg/tests/upgrade_tests/ -timescale-docker-image=$(DOCKER_IMAGE)
+
 .PHONY: go-fmt
 go-fmt:
 	go fmt ./...

```

File: Makefile [hunk 5]
```diff
@@ -8,7 +8,7 @@ build: generate
 	go build -ldflags "-X 'github.com/timescale/promscale/pkg/version.Branch=${GIT_BRANCH}' -X 'github.com/timescale/promscale/pkg/version.CommitHash=${GIT_COMMIT}'" -o dist/promscale ./cmd/promscale
 
 .PHONY: test
-test: unit e2e
+test: unit e2e upgrade_tests
```
Comment (JamesGuthrie) [inline Makefile:11]:
```suggestion
test: unit e2e upgrade-test
```

**Review by JamesGuthrie — CHANGES_REQUESTED** [review]

**Comment by cevian** [comment]
Taken over by #1283 SHOULD NOT BE MERGED

**Review by cevian — CHANGES_REQUESTED** [review]
should not be merged

**Comment by paulfantom** [comment]
I am closing this PR since it should not be merged and to stop the notifications. Please reopen if necessary.

**Pull request closed without merging**